### PR TITLE
Implemented All Requests For Voice Messaging Endpoints

### PR DESCRIPTION
**Pull request by KiPageFault**
## Checklist

- [x] Reference a related issue in the repository.
- [x] Provide a description of the proposed changes.
- [x] Ensure that code follows the project's style guidelines. Github linter checks should pass.
- [ ] Update documentation as necessary.
- [x] Verify that all tests pass.

## Reviewer Only

- [ ] **Code Quality**
  - [ ] Are variable and function names descriptive and follow naming conventions?
  - [ ] Is the code free of redundant or duplicate segments?
  - [ ] Does the code adhere to the project's style guidelines? Github linter checks shold pass.

- [ ] **Functionality**
  - [ ] Does the code perform the intended functions correctly?
  - [ ] Are all possible edge cases handled appropriately?

- [ ] **Testing**
  - [ ] Are there unit tests covering new or modified functionalities?
  - [ ] Do all existing and new tests pass successfully?
  - [ ] Are Github checks passing?

- [ ] **Documentation**
  - [Y] Is the code adequately commented to explain complex sections?
  - [Y] Have docstrings been used in functions/ methods?
  - [Y] Do docstrings follow convention throughout solution?
  - [N] Has relevant documentation been updated or created?

- [ ] **Dependencies**
  - [ ] Are new dependencies justified and documented?
  - [ ] Is the impact of these dependencies on the overall project considered?

- [ ] **Performance**
  - [ ] Is the code optimised for performance without compromising readability?

- [ ] **Security**
  - [ ] Are there any security implications introduced by the changes?
  - [ ] Have security best practices been followed?

- [ ] **Merge Preparations**
  - [ ] Is the branch up-to-date with the base branch?
  - [ ] Are there any merge conflicts that need resolution?

### CODE REVIEW

**Comment by KiPageFault** [comment]
Added All Necessary Documentation

**Review by Jordan-Prescott — CHANGES_REQUESTED** [review]
Some adjustments

File: src/odins_spear/scripter.py [hunk 1]
```diff
@@ -270,6 +270,61 @@ def user_registration(
...
+            service_provider_id,
+            group_id,
+            service_pack_id
+        )
+    
+    def vm_portal_bulk_password_set(
+        self,
+        service_provider_id: str,
+        group_id: str, 
+        new_password: str
```
[the remark this answers is not in the document]
Comment (KiPageFault) [reply src/odins_spear/scripter.py:305]: The optional password is a great idea! However for the list of users. The alteration of vm_portal_bulk_password_set in the powered by odins spear docs will allow users to perform this action themselves

File: src/odins_spear/scripts/vm_auto_login.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
+def main(api, service_provider_id: str, group_id: str, service_pack_id: str = None):
```
Comment (Jordan-Prescott) [inline src/odins_spear/scripts/vm_auto_login.py:1]: Commented on this above, the service pack id here I think we should remove.

File: src/odins_spear/endpoints/voice_messaging.py [hunk 3]
```diff
@@ -0,0 +1,581 @@
+from .base_endpoint import BaseEndpoint
```
Comment (Jordan-Prescott) [inline src/odins_spear/endpoints/voice_messaging.py:1]: Can you add in the GET POST PUT DELETE comments and make sure the methods follow the order. Nice work otherwise getting everything in!

**Review by Jordan-Prescott — CHANGES_REQUESTED** [review]
Still some parts to iron these out

File: external_call_forwarding_identifiers.csv
Comment (Jordan-Prescott) [inline external_call_forwarding_identifiers.csv:1]: This needs removing - be careful about what you commit and push. 

What's useful is having another environment on your computer where you have the latest v of OS for these kind of things.

File: src/odins_spear/endpoints/voice_messaging.py [hunk 4]
```diff
@@ -0,0 +1,574 @@
...
+
+class VoiceMessaging(
```
Comment (Jordan-Prescott) [inline src/odins_spear/endpoints/voice_messaging.py:3]: is there no POST methods in this endpoint?

File: src/odins_spear/scripter.py [hunk 5]
```diff
@@ -68,6 +68,7 @@ def bulk_password_reset(
         group_id: str,
         users: list,
         password_type: str,
+        static_password: Optional[str] = None
```
Comment (Jordan-Prescott) [inline src/odins_spear/scripter.py:71]: Have you tested this with each password type? This script updates 3 types of passwords

File: src/odins_spear/scripter.py [hunk 6]
```diff
@@ -270,7 +275,31 @@ def user_registration(
             Dict: User's ID, Device Name, and Registration status.
         """
         return self._run_script("user_registration", service_provider_id, group_id)
+    
+    def vm_auto_login(
+        self,
+        service_provider_id: str,
```
Comment (Jordan-Prescott) [inline src/odins_spear/scripter.py:281]: Can we add a param here to specify users and if no users specified we auto login all?

**Review by Jordan-Prescott — CHANGES_REQUESTED** [review]
Few bits but there are a lot of changes here so it may take us a couple of times to get it there.
